File: 7-kyu/simple-fun-#270--evil-code-medal/javascript/solution.js
const calcSeconds = (timestamp) => {
  const [ h, m, s ] = timestamp.split(":").map(Number);
  const totalSeconds = h * 3600 + m * 60 + s;
  return totalSeconds;
}

const evilCodeMedal = (userTime, ...places) => {
  const userSeconds = calcSeconds(userTime);
  const [ goldTime, silverTime, bronzeTime ] = places.map(calcSeconds);
  if (userSeconds < goldTime) {
    return "Gold";
  } else if (userSeconds < silverTime) {
    return "Silver";
  } else if (userSeconds < bronzeTime) {
    return "Bronze";
  }
  return "None";
};

File: 7-kyu/simple-fun-#270--evil-code-medal/python/solution.py
def timestamp_to_seconds(timestamp):
    h, m, s = map(int, timestamp.split(':'))
    return h * 3600 + m * 60 + s

def evil_code_medal(*args):
    user_time, gold, silver, bronze = map(timestamp_to_seconds, args)
    if user_time < gold:
        return 'Gold'
    elif user_time < silver:
        return 'Silver'
    elif user_time < bronze:
        return 'Bronze'
    return 'None'
